verification/dependencies/golang.org/x/net/internal/socket/socket.gobra

```diff
@@ -110,3 +109,28 @@ decreases
 pure func (m *Message) GetN() (res int) {
 	return unfolding acc(m.Mem(), _) in m.N
 }
+
+
+// proves injectivity of underlying buffers
+ghost
+requires 0 <= k && k < len(msgs)
+preserves acc(m.Mem(), R55)
+preserves len(m.GetFstBuffer()) > 0
+preserves sl.Bytes(m.GetFstBuffer(), 0, len(m.GetFstBuffer()))
+preserves forall i int :: { &msgs[i] } 0 <= i && i < len(msgs) ==> acc(msgs[i].Mem(), R55)
+preserves forall i int :: { &msgs[i] } 0 <= i && i < len(msgs) ==> // (VerifedSCION) TODO: Why is a  trigger like msgs[i].GetFstBuffer() breaking everything?
```

Comment:
what do you mean by "breaking everything"? Also, can you explain in a comment why this lemma is needed?

Finally, and similar to my comment on #370, can we pick a more descriptive name for the lemma?